Transcripts add temporary player item to see transcripts

SergioEstevao opened the pull request:
Refs #1848 

This PR adds a temporary tab view on the player to show a transcripts if available.

## To test

- Start the app
- Ensure that you have the `transcripts` and `newShowNotesEndpoint` FF enabled
- Open an episode with Transcripts ( Cautionary Tales podcast episode have translations)
- Play the episode
- Open the Full Screen Player
- Open the tab for Transcripts
- See if you can see the transcript and scroll around.
- Check that when the episode finishes and a new episode is loaded the transcripts updates correctly.

## Checklist

- [x] I have considered if this change warrants user-facing release notes and have added them to `CHANGELOG.md` if necessary.
- [x] I have considered adding unit tests for my changes.
- [x] I have updated (or requested that someone edit) [the spreadsheet](https://docs.google.com/spreadsheets/d/107jqrutZhU0fVZJ19SBqxxVKbV2NWSdQC9MFYdLiAxc/edit?usp=sharing) to reflect any new or changed analytics.

SergioEstevao committed:
Add Transcript view

SergioEstevao committed:
Merge branch 'feature/transcripts_add_network_parse_code' into feature/transcripts_add_temporary_player_item

SergioEstevao committed:
Add temporary transcripts tabs on episode player in order to show transcripts

dangermattic commented:
<!--
  0 Errors
  1 Warning: View files have been modified,...
  0 Messages
  0 Markdowns
-->
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">View files have been modified, but no screenshot or video is included in the pull request. Consider adding some for clarity.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

SergioEstevao committed:
Format text in transcript

SergioEstevao committed:
Show scroll indicator correctly

SergioEstevao committed:
Add a loading indicator and update transcripts when changing episode

SergioEstevao committed:
Show failed loads and no transcripts

SergioEstevao commented on podcasts/PlayerTabsView.swift:
```diff
@@ -21,6 +22,8 @@ enum PlayerTabs: Int {
         case .bookmarks:
             return L10n.bookmarks
+        case .transcripts:
+            return "Transcripts"
```
I'm not localising the strings on this PR because these are temporary until we have final designs.

leandroalonso replied:
I believe we can start doing so, unless is a very specific copy? "Transcripts" I'm sure we will have in the final version somehow.

Anyway, not a strong feeling.

SergioEstevao committed:
Update colors

SergioEstevao committed:
Merge branch 'feature/transcripts_add_network_parse_code' into feature/transcripts_add_temporary_player_item

SergioEstevao committed:
Remove unnecessary dispatch to main queue

Because the class is derived from ViewController is already marked as main actor

leandroalonso reviewed (approved):
Tested with the flag enabled and disabled and it works as expected.

@SergioEstevao one logic we might want to add is to display the tab only if a transcript is ready. This is a logic we will need anyway, independently of design: if there are transcripts we will show a tab/card/button/something, if not, we won't show.

However, I don't feel this needs to be tackled now, just something for us to keep in mind (and this is kinda done in the transcripts VC).

SergioEstevao commented:
> However, I don't feel this needs to be tackled now, just something for us to keep in mind (and this is kinda done in the transcripts VC).

I will add this on the follow up PR

SergioEstevao committed:
Localize Transcript string

SergioEstevao committed:
Merge branch 'trunk' into feature/transcripts_add_temporary_player_item

# Conflicts:
#	podcasts/en.lproj/Localizable.strings
